### PR TITLE
fix minor issues, add volatile mutex for #184

### PR DESCRIPTION
**Pull request by kinvaris**

### CODE REVIEW

**Comment by JeffreyDevloo** [comment]
The implementation for volatile mutex is right. Locking certain checks to only run once on the cluster.
However this is just implement the request of #184 partially. The whole healthcheck can still be run parallel of each other on the same node.
Add file mutex to all high level execution methods. (/lib/healthcheck.py methods)

File: ovs/extensions/healthcheck/alba/alba_health_check.py [hunk 1]
```diff
@@ -171,84 +173,97 @@ def check_if_proxies_work(logger):
...
-                                        .hexdigest()
-                                    hash_fetched = hashlib.md5(open(AlbaHealthCheck.TEMP_FILE_FETCHED_LOC, 'rb')
-                                                               .read()).hexdigest()
-
-                                    if hash_original == hash_fetched:
-                                        logger.success("Creation of a object in namespace '{0}' on proxy '{1}' "
-                                                       "with preset '{2}' succeeded!"
-                                                       .format(namespace_key,service.name,preset.get('name')),
-                                                       '{0}_preset_{1}_compare_object'
+                                with volatile_mutex('ovs-healthcheck_proxy-test'):
```
Comment (JeffreyDevloo) [inline ovs/extensions/healthcheck/alba/alba_health_check.py]: Why lock this method cluster wide?
This test is testing its own proxy.

File: ovs/extensions/healthcheck/arakoon/arakooncluster_health_check.py [hunk 2]
```diff
@@ -325,45 +330,46 @@ def verify_integrity(logger, arakoon_clusters=None):
             if ArakoonHealthCheck.MACHINE_DETAILS.machine_id not in cluster_info:
                 continue
 
-            tries = 1
-            max_tries = 2  # should be 5 but .nop is taking WAY to long
+            with volatile_mutex('ovs-healthcheck_arakoon-test_{0}'.format(cluster_name)):
```
Comment (JeffreyDevloo) [inline ovs/extensions/healthcheck/arakoon/arakooncluster_health_check.py]: Arakoon collapse test tries to collapse local arakoons. By locking them cluster wide, you will have only tested the arakoons on this node and will report a false value back to the consumer.

File: ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py [hunk 3]
```diff
@@ -127,56 +139,232 @@ def check_volumedrivers(logger):
...
-                                                       VolumedriverHealthCheck.MACHINE_ID))
-                            # create a new one
-                            if VolumedriverHealthCheck._check_volumedriver(name, storagedriver_guid, vp.name):
-                                # delete the recently created
-                                if VolumedriverHealthCheck._check_volumedriver_remove(vpool_name=vp.name,
-                                                                                      vdisk_name=name):
-                                    # working
-                                    logger.success("Volumedriver of vPool '{0}' is working fine!".format(vp.name),
-                                                   'volumedriver_{0}'.format(vp.name))
+                with volatile_mutex('ovs-healthcheck_check-volumedrivers_{0}'
```
Comment (JeffreyDevloo) [inline ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py]: This test only tests the local volumedriver. Don't lock it for the whole cluster.

File: ovs/extensions/healthcheck/arakoon/arakooncluster_health_check.py (hunk 2 above)
Comment (kinvaris) [inline ovs/extensions/healthcheck/arakoon/arakooncluster_health_check.py]: Agreed with using a file mutex but this is not the arakoon collapse test, it's the arakoon integrity test

**Pull request merged**